src/vcSHP.cpp

```diff
@@ -0,0 +1,741 @@
+  }
+  else
+  {
+    nodeName = udTempStr("POI %d", record.id);
+  }
+
+  udProjectNode *pNode = nullptr;
+  udProjectNode_Create(pProgramState->activeProject.pProject, &pNode, pParentNode, "POI", nodeName, nullptr, nullptr);
+
+  switch (record.shapeType)
+  {
+  case vcSHP_Type::Point:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.point, 1);
+  break;
+  case vcSHP_Type::Polyline:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.poly.points, record.data.poly.pointsCount);
+  break;
+  case vcSHP_Type::Polygon:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.poly.points, record.data.poly.pointsCount);
+  break;
+  case vcSHP_Type::MultiPoint:
+  vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHP_Type::PointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointZ.point, 1);
+  break;
+  case vcSHP_Type::PolylineZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHP_Type::PolygonZ:
+  case vcSHP_Type::MultiPatch:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHP_Type::MultiPointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHP_Type::PointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointM.point, 1);
+  break;
+  case vcSHP_Type::PolylineM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHP_Type::PolygonM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHP_Type::MultiPointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointM.points, record.data.multiPointM.pointsCount);
+  break;
+  default:
+    printf("invalid shape type %d\n", (int)record.shapeType);
```

Comment:
This isn't necessary and should be removed.

Reply:
fixed.